language/stdlib/modules/DesignatedDealer.move

```diff
@@ -105,7 +105,8 @@ module DesignatedDealer {
         blessed: &signer, amount: u64, addr: address, tier_index: u64
     ): Libra<CoinType> acquires Dealer {
         Association::assert_account_is_blessed(blessed);
-
+        // INVALID TIER_INDEX (if tier is 4, can mint unlimited)
+        assert(tier_index <= 4, 66);
```

Comment:
I would feel better if this check + the `amount` check were in `tiered_mint_` to prevent a different caller from forgetting to check them. This might also let you get rid of the `tier_index >= number_of_tiers` check in `tiered_mint`.